### PR TITLE
Romanchuk / Upgrade EF Core to 7

### PR DESCRIPTION
**Pull request by alrom**

### CODE REVIEW

File: OutOfSchool/OutOfSchool.DataAccess/OutOfSchool.DataAccess.csproj [hunk 1]
```diff
@@ -28,12 +28,12 @@
       <PackageReference Include="H3Lib" Version="3.7.2" />
       <PackageReference Include="Microsoft.AspNetCore.DataProtection.EntityFrameworkCore" Version="6.0.6" />
       <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="6.0.6" />
-      <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.6" />
-      <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="6.0.6">
+      <PackageReference Include="Microsoft.EntityFrameworkCore" Version="7.0.2" />
```
Comment (h4wk13) [inline OutOfSchool/OutOfSchool.DataAccess/OutOfSchool.DataAccess.csproj:31]: Have you read:
[https://learn.microsoft.com/en-us/ef/core/what-is-new/ef-core-7.0/breaking-changes](https://learn.microsoft.com/en-us/ef/core/what-is-new/ef-core-7.0/breaking-changes)
It describes a few high impact changes.

Comment (alrom) [reply OutOfSchool/OutOfSchool.DataAccess/OutOfSchool.DataAccess.csproj:31]: Thx. I don't know how to change this PR to draft after started it to review )))

Comment (alrom) [reply OutOfSchool/OutOfSchool.DataAccess/OutOfSchool.DataAccess.csproj:31]: I am reading the article and i'll make some changes. Thx again.

File: OutOfSchool/OutOfSchool.DataAccess/Repository/NotificationRepository.cs [hunk 2]
```diff
@@ -25,17 +25,11 @@ public NotificationRepository(OutOfSchoolDbContext dbContext)
...
-        await db.Database.ExecuteSqlRawAsync(
-            @"DELETE FROM Notifications WHERE ReadDateTime IS NULL AND CreatedDateTime < {0};
-            DELETE FROM Notifications WHERE ReadDateTime IS NOT NULL AND CreatedDateTime < {1};",
-            dateNotReaded,
-            dateReaded);
+        var dateWasNotRead = DateTimeOffset.UtcNow.AddYears(-1);
+        var dateWasRead = DateTimeOffset.UtcNow.AddMonths(-1);
+
+        await db.Notifications.Where(x => x.ReadDateTime == null && x.CreatedDateTime < dateWasNotRead).ExecuteDeleteAsync();
+        await db.Notifications.Where(x => x.ReadDateTime != null && x.CreatedDateTime < dateWasRead).ExecuteDeleteAsync();
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.DataAccess/Repository/NotificationRepository.cs:32]: (Maybe performance) I wonder why this couldn't be done in one query? I mean usage of 
```
    .Where(x => (x.ReadDateTime == null && x.CreatedDateTime < dateWasNotRead) 
        || (x.ReadDateTime != null && x.CreatedDateTime < dateWasRead)) 
```

File: OutOfSchool/OutOfSchool.DataAccess/OutOfSchool.DataAccess.csproj [hunk 3]
```diff
@@ -28,12 +28,12 @@
...
+      <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="7.0.2">
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.DataAccess/OutOfSchool.DataAccess.csproj:32]: (Same version for all references) Shouldn't specific version be set using solution-level `Directory.Packages.props`?

File: OutOfSchool/OutOfSchool.DataAccess/Repository/NotificationRepository.cs (hunk 2 above)
Comment (alrom) [reply OutOfSchool/OutOfSchool.DataAccess/Repository/NotificationRepository.cs:32]: It was only for readability.

Comment (VadymLevkovskyi) [reply OutOfSchool/OutOfSchool.DataAccess/Repository/NotificationRepository.cs:32]: Now we have (sligtly better) readability with worse performance. Is it worth it?

**Pull request closed without merging**